### ISSUE TITLE
Firelocks have lost their y-sorting ability

### ISSUE THREAD

**Issue by SweptWasTaken** [space-wizards/space-station-14#5787]
## Description

Not really sure which PR broke this but y-sorting for fire-lock frames isn't working anymore.

I seem to remember this working pretty flawlessly a while back so it's a shame it's broke.

**Screenshots**

![Content Client_2021-12-13_18-11-49](https://user-images.githubusercontent.com/49448379/146076235-880816c5-e8e8-4c9e-8b11-57304a1097fc.png)

**Additional context**

**Comment by ElectroJr** [comment]
The draw depth for open firelocks were moved down so that you could have firelocks on tables & windoors and still be able to interact with them. See #5120 or these  images:

before:
![a](https://user-images.githubusercontent.com/60421075/146109862-a85fc59b-e11b-4532-ab27-3a409086bdbc.png)

after:
![b](https://user-images.githubusercontent.com/60421075/146109873-dd796c85-2c4d-41ec-81db-6fc875ba8f46.png)

This does mean they now appear under humans but IMO is worth it to be able to use them with windoors.

Though looking back at it now, maybe the right choice would have been to adjust the draw depth for only some of the firelocks.
The open & closed draw-depths for doors is now just a yaml datafield, so you could just make a slight variation of firelocks that have a lower drawdepth while open, and then revert the other firelocks back to their original draw depth.

**Comment by SweptWasTaken** [comment]
> The draw depth for open firelocks were moved down so that you could have firelocks on tables & windoors and still be able to interact with them. See #5120 & the mentioned pr, or just these images:
> 
> before: ![a](https://user-images.githubusercontent.com/60421075/146109862-a85fc59b-e11b-4532-ab27-3a409086bdbc.png)
> 
> after: ![b](https://user-images.githubusercontent.com/60421075/146109873-dd796c85-2c4d-41ec-81db-6fc875ba8f46.png)
> 
> This does mean they now appear under humans but IMO is worth it to be able to use them with windoors.
> 
> Though looking back at it now, maybe the right choice would have been to adjust the draw depth for only some of the firelocks. The open & closed draw-depths for doors is now just a yaml datafield, so you could just make a slight variation of firelocks that have a lower drawdepth while open, and then revert the other firelocks back to their original draw depth.

Yeah I don't think that should have been merged, it's essentially trading functionality without fixing the issue.

**Comment by PaulRitter** [comment]
notabug & wontfix but unironically.

everything became more interactable this way, so there either needs to be a replacement-solution or this stays.

eitherway, this isnt an issue afaict

**Comment by SweptWasTaken** [comment]
> notabug & wontfix but unironically.
> 
> everything became more interactable this way, so there either needs to be a replacement-solution or this stays.
> 
> eitherway, this isnt an issue afaict

I think it's still an issue because a feature of firelock frames has been passed over instead of compromised upon. 

Wouldn't a solution for interactability and keeping the y-sorting be easy to implement? Maybe add a check for windoors or tables on the same tiles as firelocks, or have separate firelock entities for windows and general station areas?

**Comment by PaulRitter** [comment]
i still dont see the issue, sorry
